src/Language/Fixpoint/Solver/Interpreter.hs

```diff
@@ -503,18 +511,20 @@ interpret ie γ ctx env   (ETApp e1 t)   = let e1' = interpret' ie γ ctx env e1
 interpret ie γ ctx env   (ETAbs e1 sy)  = let e1' = interpret' ie γ ctx env e1 in ETAbs e1' sy
 interpret ie γ ctx env   (PAnd es)      = let es' = map (interpret' ie γ ctx env) es in go [] (reverse es')
   where
-    go []  []     = PTrue
-    go [p] []     = interpret' ie γ ctx env p
-    go acc []     = PAnd acc
-    go acc (e:es) = if e == PTrue then go acc es
-                                  else if e == PFalse then PFalse else go (e:acc) es
+    go []  []         = PTrue
+    go [p] []         = interpret' ie γ ctx env p
+    go acc []         = PAnd acc
+    go acc (PTrue:es) = go acc es
+    go _   (PFalse:_) = PFalse
+    go acc (e:es)     = go (e:acc) es
```

Comment:
Purely a stylistic refactoring. I think this is easier to read, but I'm happy to revert this change if there is opposition.